### PR TITLE
Comments endpoint serving Get request Issue #22

### PR DESCRIPTION
**Pull request by ahk4815**
Fixes: #22 
A comments endpoint is added which serves HTTPS Get requests. A comments.ts file is made where the logic for handling get requests is done. The demo data is being fetched from a comment.json file.

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -1,7 +0,0 @@
-DB_CONNECTION_URL=From MongoDB Atlas Panel > CONNECT > Connect Your Application > Connection String
-USE_DOCKER= true
-MONGO_INITDB_ROOT_USERNAME= admin1
-MONGO_INITDB_ROOT_PASSWORD= adminPassword
-MONGO_INITDB_DATABASE= RESTfetchdb
-MONGO_NON_ROOT_USERNAME= user1
-MONGO_NON_ROOT_PASSWORD= userPassword
```
Comment (ShauryaAg) [inline .env.example:7]: Why was this file deleted?

**Review by ShauryaAg — CHANGES_REQUESTED** [review]

File: .env.example (hunk 1 above)
Comment (ahk4815) [reply .env.example:7]: > Why was this file deleted?

It was deleted by mistake since .env was already made by me, i didnt notice this one. My apologies!
I am trying if i can add this file again in this PR

Comment (ahk4815) [reply .env.example:7]: > > Why was this file deleted?
> 
> It was deleted by mistake since .env was already made by me, i didnt notice this one. My apologies!
> I am trying if i can add this file again in this PR

I have added the file by copy pasting from original repo, will that work

**Review by ShauryaAg — APPROVED** [review]

**Review by paramsiddharth — APPROVED** [review]

**Comment by adityabisoi** [comment]
Please resolve conflicts. The `_id` fields are not needed, so please remove them. Also, please change the field name from `comments` to `data`

File: src/api/routes/comments.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+//endpoint for /GET requests for comments
+router.get('/', (req: Request, res: Response, next: NextFunction) => {
+//reading the contents of comments json file
+fs.readFile('./src/api/routes/comment.json', (err: string, data: string) => {
+    if (err) {
+      console.log(err)
+      res.status(500).json({
+          error: err
+      })
```
Comment (paramsiddharth) [inline src/api/routes/comments.ts:14]: Instead of simply calling `res.status(...).json(...)`, make sure to return it as well to make sure `res.json` isn't called twice i. e. The function terminates if an error occurs.
```js
return res.status(500).json({
    error: err
})
```

File: src/api/routes/comments.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+// This endpoint supports fetching comments with their id
+router.get('/:commentId', (req: Request, res: Response, next: NextFunction) => {
+    const id: string = req.params.commentId
+    fs.readFile('./src/api/routes/comment.json', (err: string, data: string) => {
+        if (err) {
+          console.log(err)
+          res.status(500).json({
+              error: err
+          })
```
Comment (paramsiddharth) [inline src/api/routes/comments.ts:31]: Do the same as mentioned above.

**Review by paramsiddharth — CHANGES_REQUESTED** [review]

**Comment by ahk4815** [comment]
> Please resolve conflicts. The `_id` fields are not needed, so please remove them. Also, please change the field name from `comments` to `data`

**Comment by ahk4815** [comment]
> Please resolve conflicts. The `_id` fields are not needed, so please remove them. Also, please change the field name from `comments` to `data`

If the _id field is removed, then the Get request for fetching comments with their ids will also have to be removed. So I am removing the get CommentById request as well

File: src/api/routes/comments.ts (hunk 2 above)
Comment (ahk4815) [reply src/api/routes/comments.ts]: I have added the return statement. Please inform if that was what you were asking.

**Comment by ahk4815** [comment]
> Please resolve conflicts. The `_id` fields are not needed, so please remove them. Also, please change the field name from `comments` to `data`

The conflicts are resolved. _id field is removed, 'comments' field is changed to 'data'

File: src/api/routes/comments.ts (hunk 2 above)
Comment (ahk4815) [reply src/api/routes/comments.ts]: res.status() is now being returned in the function

File: src/api/routes/comments.ts (hunk 3 above)
Comment (ahk4815) [reply src/api/routes/comments.ts]: lines 29,30,31 are deleted so i dont think i  have to add anything there

**Review by adityabisoi — APPROVED** [review]

**Comment by paramsiddharth** [comment]
@ahk4815 Please resolve the conflicts!

**Comment by ahk4815** [comment]
okay i am working on it

**Comment by ahk4815** [comment]
> @ahk4815 Please resolve the conflicts!

I have resolved the conflicts, have a look

**Review by paramsiddharth — APPROVED** [review]

**Comment by paramsiddharth** [comment]
Thank you for the contribution!
